### ISSUE TITLE
[stylelint] improvements to stylelint rule (docusaurus/copyright-header)

### ISSUE THREAD

**Issue by yangshun** [facebook/docusaurus#2370]
## 💥 Proposal

This is a follow up of #2361 and #2363 regarding improving our stylelint test (`packages/stylelint-copyright`)

- It is using the Tape test runner and is not integrated into our CI. If we move the implementation to Jest, it'll be run as part of the repo root's `yarn test` flow which is part of CI and that'll be great. stylelint's rules are [already using Jest](https://github.com/stylelint/stylelint/blob/master/lib/rules/at-rule-blacklist/__tests__/index.js) and perhaps we can learn from how they do it
- Allow for rule configuration options to specify the copyright pattern similar to [`eslint-plugin-header`](https://github.com/Stuk/eslint-plugin-header)

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

**Comment by ayshiff** [comment]
I would love to work on it @yangshun 👍

**Comment by yangshun** [comment]
@ayshiff go ahead

**Comment by ayshiff** [comment]
Hey @yangshun, I have a question about the way stylelint rules should be tested !

I followed the way stylelint's rules were tested by adding a **jest setup file** and defining a **testRule method**.
I think I have to do some tricks around the way the plugin is created because it gives me the error  "Unknown rule docusaurus/copyright-header."

Otherwise I checked the way "stylelint-test-rule-tape" was working and it uses the version « 9.1.1 » of stylelint where we can access a createRuleTester method to make our own rule tester. We can then create our own assertEquality function using Jest.

I think the best way is to follow the first method but I wanted to have your opinion before.
